### DISCUSSION TITLE
createSignedUrls is missing api url

### DISCUSSION THREAD

**Discussion by Ben1980** [supabase/supabase#13153] [Questions] [answered]
Why is createSignedUrls not including the api url as createSignedUrl does? Is there a reason for this?

```
Future<String> createSignedUrl(
    String path,
    int expiresIn, {
    TransformOptions? transform,
  }) async {
    final finalPath = _getFinalPath(path);
    final options = FetchOptions(headers: headers);
    final response = await storageFetch.post(
      '$url/object/sign/$finalPath',
      {
        'expiresIn': expiresIn,
        if (transform != null) 'transform': transform.toQueryParams,
      },
      options: options,
    );
    final signedUrlPath = (response as Map<String, dynamic>)['signedURL'];
    final signedUrl = '$url$signedUrlPath'; <-- Here I understand $url is the API url
    return signedUrl;
  }
```
```
Future<List<SignedUrl>> createSignedUrls(
    List<String> paths,
    int expiresIn,
  ) async {
    final options = FetchOptions(headers: headers);
    final response = await storageFetch.post(
      '$url/object/sign/$bucketId',
      {
        'expiresIn': expiresIn,
        'paths': paths,
      },
      options: options,
    );
    final List<SignedUrl> urls = (response as List).map((e) {
      return SignedUrl(
        path: e['path'],
        signedUrl: e['signedURL'], <-- Here the API URL is missing
      );
    }).toList();
    return urls;
  }

```

This way it would work:
`signedUrl: url + (e['signedURL'] as String),`
<img width="503" alt="Bildschirm­foto 2023-03-22 um 15 40 57" src="https://user-images.githubusercontent.com/919260/226940273-26b6c1f5-0c42-4b14-8932-daea8e67ca5b.png">

**Comment by GaryAustin1** [comment, marked as answer]
Not sure if you mean no URL at all.   You are showing using ‘signedURL’ with caps in your 2 case you say does not work.

**Reply by Ben1980** [reply]
Hi,
createSignedUrl is returning the full Url like: http://localhost:...../object/sign/....
while createSignedUrls is just returning inside signedUrl: /object/sign/.....
That's a bit surprising, I would expect that both functions return the same URL. Or do I misunderstanding something?

**Reply by GaryAustin1** [reply]
You need to use `signedUrl: e['signedUrl'], `  not with signedURL (which just returns the object name).  The caps in URL make a difference.

**Reply by Ben1980** [reply]
I'm not sure we talk about the same thing. The code above is actually the functionality provided by flutter_supabase. I just call it. And there the returned list of ist not containing the full URL if function createSignedUrls is used. But for createSignedUrl it is working.
Maybe I have to ask this in the respective repository.

**Reply by GaryAustin1** [reply]
You need to change `signedUrl: e['signedURL']` to `signedUrl: e['signedUrl']`
See in this pix how there is both signedURL AND signedUrl with the full path....

![image](https://user-images.githubusercontent.com/54564956/227232276-a6a0eef6-b829-473f-b76a-f5be97f05128.png)

**Reply by bmahrkso** [reply]
Thats JS isn't it? I think i found the difference.
in storage-JS you have
createSignedUrl:
`const signedUrl = encodeURI(`${this.url}${data.signedURL}${downloadQueryParam}`)`
createSignedUrls
`encodeURI(`${this.url}${datum.signedURL}${downloadQueryParam}`)`
So in principal they are identical.
In storage-dart unfortunately they differ:
createSignedUrl:
` final signedUrl = '$url$signedUrlPath';`
createSignedUrls
` signedUrl: e['signedURL'],`
and this one is not giving the Url.
I can't check right now because i'm at work, also the reason for the different gitHub account, but i will chekc as soon as possible.

**Reply by GaryAustin1** [reply]
You can't change signedURL to signedUrl in  `signedUrl: e['signedURL']` ?  Sorry I don't know Dart/Flutter but if you can't then sounds like an issue for that repository as both URL and Url types are available to it from the storage-api response.

**Reply by Ben1980** [reply]
I made a pull request in Storage-dart fixing this issue. Now it's working as expected
